uc-crux-llvm/src/UCCrux/LLVM/Config.hs

```diff
@@ -0,0 +1,176 @@
+{-
+Module       : UCCrux.LLVM.Config
+Description  : CLI, environment, and config file options
+Copyright    : (c) Galois, Inc 2021
+License      : BSD3
+Maintainer   : Langston Barrett <langston@galois.com>
+Stability    : provisional
+-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module UCCrux.LLVM.Config
+  ( UCCruxLLVMOptions (..),
+    ucCruxLLVMConfig,
+    processUCCruxLLVMOptions,
+  )
+where
+
+{- ORMOLU_DISABLE -}
+import           Control.Applicative ((<|>))
+import           Control.Monad (when)
+import           Data.Functor ((<&>))
+
+import qualified Crux.Config as Crux
+import           Crux.Config.Common (CruxOptions, loopBound, recursionBound)
+import           Crux.LLVM.Config (LLVMOptions, llvmCruxConfig)
+import           CruxLLVMMain (processLLVMOptions)
+
+import           UCCrux.LLVM.Logging (verbosityFromInt)
+import           UCCrux.LLVM.Context.App (AppContext, makeAppContext)
+{- ORMOLU_ENABLE -}
+
+data UCCruxLLVMOptions = UCCruxLLVMOptions
+  { ucLLVMOptions :: LLVMOptions,
+    doExplore :: Bool,
+    reExplore :: Bool,
+    entryPoints :: [String],
+    exploreBudget :: Int,
+    skipFunctions :: [String],
+    verbosity :: Int
+  }
+
+processUCCruxLLVMOptions ::
+  (CruxOptions, UCCruxLLVMOptions) -> IO (AppContext, CruxOptions, UCCruxLLVMOptions)
+processUCCruxLLVMOptions (initCOpts, initUCOpts) =
+  do
+    let appCtx = makeAppContext (verbosityFromInt (verbosity initUCOpts))
+    when (not (doExplore initUCOpts) && null (entryPoints initUCOpts)) $
+      fail "At least one entry point (--entry-points) is required (or try --explore)"
```

Comment:
I'm generally skeptical of `fail` these days. Is this a `System.Exit.die` kind of condition?

Reply:
Yeah, I think that's basically equivalent since this is just in IO. What are the pros/cons of using `System.Exit.die` over `fail` here?

Reply:
I just like being explicit about the nature of the failure (i.e., it isn't supposed to be recoverable). You can also see `die` and not have to check what monad you are in to determine how `fail` will be interpreted. Possibly good or bad, depending on your point of view: `die` spews its output to stderr before it throws its exception.

Reply:
Noted, I'll try and keep that in mind as a practice in the future (and I'll change it here)!